### PR TITLE
Refactor Log SDK to implement OTEP-0150

### PR DESCRIPTION
**Pull request by jack-berg**
Refactor log sdk to implement [OTEP-0150](https://github.com/open-telemetry/oteps/blob/main/text/logs/0150-logging-library-sdk.md).

Also brings it much more in line with the patterns of traces and metrics. 

Resolves Issue #3750. 

I still need to write some tests for some of the components added.

### CODE REVIEW

**Comment by jack-berg** [comment]
> I'm currently working to align this with OTEl 0150. No need to review until that's done.

Done. I'm updating the PR title and description accordingly.

**Comment by jack-berg** [comment]
This PR is ready for review.

The log SDK now reflects [OTELP-0150](https://github.com/open-telemetry/oteps/blob/main/text/logs/0150-logging-library-sdk.md), and has patterns that closely mirrors the trace SDK.

If we could get these changes into version 1.8.0, we could start to have some appenders contributed to `opentelemetry-java-instrumentation` that leverage it, like the [log4j2 appender prototype](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/4375).

**Comment by jack-berg** [comment]
@jkwatson I just pushed a commit to switch to the `LogEmitter.logBuilder().emit()` pattern we've discussed. I can remove the commit and do it in a separate PR, but I figured that it was best to do it here because this work is predominantly concerned with the flow of using the SDK, and that type of change has repercussions on the flow.

**Comment by jkwatson** [comment]
> @jkwatson I just pushed a commit to switch to the `LogEmitter.logBuilder().emit()` pattern we've discussed. I can remove the commit and do it in a separate PR, but I figured that it was best to do it here because this work is predominantly concerned with the flow of using the SDK, and that type of change has repercussions on the flow.

no, that's fine. no worries. Just thought we should get this thing merged and tweak from there.

File: sdk/logs/src/main/java/io/opentelemetry/sdk/logs/data/LogRecord.java [hunk 1]
```diff
@@ -5,47 +5,80 @@
...
-        body,
-        attributes);
-  }
+
+/**
+ * Implementation of a log as defined in <a
+ * href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md">OpenTelemetry
+ * Log Data Model</a>.
+ */
+public interface LogRecord {
```
[the remark this answers is not in the document]
Comment (lonewolf3739) [reply sdk/logs/src/main/java/io/opentelemetry/sdk/logs/data/LogRecord.java]: @tigrannajaryan There was nothing odd about the LogRecord and it is mostly similar to [`LogRecord`](https://docs.python.org/3/library/logging.html#logging.LogRecord) from python standard library. In Python SDK we just do one-to-one [mapping](https://github.com/open-telemetry/opentelemetry-python/blob/logs/opentelemetry-sdk/src/opentelemetry/sdk/_logs/__init__.py#L321) from the stdlib log record to OTEP model.

[the remark this answers is not in the document]
Comment (tigrannajaryan) [reply sdk/logs/src/main/java/io/opentelemetry/sdk/logs/data/LogRecord.java]: Submitted an amendment to OTEP: https://github.com/open-telemetry/oteps/pull/183

**Pull request merged**